udev_monitor.c

```diff
@@ -116,25 +114,23 @@ static void *udev_monitor_handle_event(void *ptr)
 {
     struct udev_monitor *udev_monitor = ptr;
     struct inotify_event *event;
-    struct epoll_event epoll[2];
+    struct pollfd fds[2];
     char data[4096];
-    sigset_t set;
     ssize_t len;
     int i;
 
-    sigfillset(&set);
-    pthread_sigmask(SIG_BLOCK, &set, NULL);
+    fds[0] = (struct pollfd) {udev_monitor->ifd, POLLIN};
+    fds[1] = (struct pollfd) {udev_monitor->pfd[0], POLLIN};
 
-    while (epoll_wait(udev_monitor->efd, epoll, 2, -1) != -1) {
-        for (i = 0; i < 2; i++) {
-            if (epoll[i].data.fd != udev_monitor->pfd[0]) {
-                continue;
-            }
+    while (1) {
+        if (poll(fds, 2, -1) == -1) {
+            continue;
+        }
 
+        if (fds[1].revents & POLLIN) {
             read(udev_monitor->pfd[0], data, 1);
             write(udev_monitor->pfd[1], "1", 1);
```

Comment:
We need to remove this synchronization as soon as THREADS_MAX is removed. We don't need THREADS_MAX, we only need one thread with event loop.

Reply:
I don't disagree. Was going to send out another PR, but yeah can just add to this one. 

I was slow to send this message and it is already outdated so will just post the rest of what I was going to say as its own comment.